hack/generator/pkg/jsonast/schema_abstraction.go

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package jsonast
+
+import (
+	"net/url"
+)
+
+// Schema abstracts over the exact implementation of
+// JSON schema we are using as we need to be able to
+// process inputs provided by both 'gojsonschema' and
+// 'go-openapi'. It is possible we could switch to
+// 'go-openapi' for everything because it could handle
+// both JSON Schema and Swagger; but this is not yet done.
+type Schema interface {
+	URL() *url.URL
+	Title() *string
+	Description() *string
+
+	HasType(schemaType SchemaType) bool
+
+	// complex things
```

Comment:
Do these comments get interpreted as documentation comments for the field which they are directly above?

Not sure if there's a way to avoid that but worth considering.

Reply:
I think you have to start them with the name of the field.